src/createRollupPrepareTransactionRequest.ts

```diff
@@ -48,13 +73,30 @@ export async function createRollupPrepareTransactionRequest<TChain extends Chain
 
   if (isNonZeroAddress(params.nativeToken)) {
     const isRollupChain = chainConfig.arbitrum.DataAvailabilityCommittee === false;
-    const isFeeTokenPricerMissing = !isNonZeroAddress(params.feeTokenPricer);
 
-    // fee token pricer is mandatory for custom gas token rollup chains
-    if (isRollupChain && isFeeTokenPricerMissing) {
-      throw new Error(
-        `"params.feeTokenPricer" must be provided for a custom gas token rollup chain.`,
-      );
+    switch (rollupCreatorVersion) {
+      case 'v2.1': {
+        if (isRollupChain) {
+          throw new Error(
+            `"params.nativeToken" can only be used on AnyTrust chains. Set "arbitrum.DataAvailabilityCommittee" to "true" in the chain config.`,
+          );
+        }
+
+        break;
+      }
+
+      case 'v3.1': {
+        const isFeeTokenPricerMissing = !isNonZeroAddress(
+          (params as CreateRollupParams<'v3.1'>).feeTokenPricer,
+        );
+
+        // fee token pricer is mandatory for custom gas token rollup chains
+        if (isRollupChain && isFeeTokenPricerMissing) {
+          throw new Error(
+            `"params.feeTokenPricer" must be provided for a custom gas token rollup chain.`,
+          );
+        }
+      }
     }
```

Comment:
some validation that was specific to v2.1 and v3.1, didn't want to have separate files just for this little of difference. we still have full test files for both v2.1 and v3.1